Modules/_testcapi/vectorcall.c

```diff
@@ -102,6 +102,23 @@ test_pyobject_vectorcall(PyObject *self, PyObject *args)
     return PyObject_Vectorcall(func, stack, nargs, kwnames);
 }
 
+static PyObject *
+override_vectorcall(
+    PyObject *callable, PyObject *const *args, size_t nargsf, PyObject *kwnames) {
+    return PyUnicode_FromString("overridden");
+}
+
+static PyObject *
+function_setvectorcall(PyObject *self, PyObject *func)
+{
+    if (!PyFunction_Check(func)) {
+        PyErr_BadInternalCall();
```

Comment:
This should be a TypeError.

```suggestion
        PyErr_SetString(PyExc_TypeError, "'func' must be a function");
```